providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithClientConfigurer.java

```diff
@@ -0,0 +1,167 @@
+            throw new InvalidOptions("Flagsmith API key has not been set.");
+        }
+
+        if (options.getEnvFlagsCacheKey() == null
+            && (options.getExpireCacheAfterWrite() != 0
+            || options.getExpireCacheAfterAccess() != 0
+            || options.getMaxCacheSize() != 0
+            || options.isRecordCacheStats())) {
+            throw new InvalidCacheOptions(
+                "No Flagsmith cache key provided but other cache settings have been set."
+            );
+        }
+    }
+
+    /**
+     * initializeProvider is initializing the different class element used by the provider.
+     *
+     * @param options the options used to create the provider
+     */
+    static FlagsmithClient initializeProvider(FlagsmithProviderOptions options) {
+        FlagsmithClient.Builder flagsmithBuilder = FlagsmithClient
+            .newBuilder();
+
+        // Set main configuration settings
+        if (options.getApiKey() != null) {
+            flagsmithBuilder.setApiKey(options.getApiKey());
+        }
+
+        if (options.getHeaders() != null && !options.getHeaders().isEmpty()) {
+            flagsmithBuilder.withCustomHttpHeaders(options.getHeaders());
+        }
+
+        if (options.getEnvFlagsCacheKey() != null) {
+            FlagsmithCacheConfig flagsmithCacheConfig = initializeCacheConfig(options);
+            flagsmithBuilder.withCache(flagsmithCacheConfig);
+        }
+
+        com.flagsmith.config.FlagsmithConfig flagsmithConfig = initializeConfig(options);
+        flagsmithBuilder.withConfiguration(flagsmithConfig);
+
+        return flagsmithBuilder.build();
+    }
+
+    /**
+     * Sets the cache related configuration for the provider using
+     * the FlagsmithCacheConfig builder.
+     *
+     * @param options the options used to create the provider
+     * @return a FlagsmithCacheConfig object containing the FlagsmithClient cache options
+     */
+    private static FlagsmithCacheConfig initializeCacheConfig(FlagsmithProviderOptions options) {
+        FlagsmithCacheConfig.Builder flagsmithCacheConfig = FlagsmithCacheConfig.newBuilder();
+
+        // Set cache configuration settings
+        if (options.getEnvFlagsCacheKey() != null) {
+            flagsmithCacheConfig.enableEnvLevelCaching(options.getEnvFlagsCacheKey());
+        }
+
+        if (options.getExpireCacheAfterWrite() != 0
+            && options.getExpireCacheAfterWriteTimeUnit() != null) {
+            flagsmithCacheConfig.expireAfterAccess(
+                options.getExpireCacheAfterWrite(),
+                options.getExpireCacheAfterWriteTimeUnit());
+        }
+
+        if (options.getExpireCacheAfterAccess() != 0
+            && options.getExpireCacheAfterAccessTimeUnit() != null) {
+            flagsmithCacheConfig.expireAfterAccess(
+                options.getExpireCacheAfterAccess(),
+                options.getExpireCacheAfterAccessTimeUnit());
+        }
+
+        if (options.getMaxCacheSize() != 0) {
```

Comment:
The default is zero and comes from `FlagsmithProviderOptions` (primitive default). Should we allow a zero cache ? IMO this is misleading and a non-zero default makes sense

Reply:
Agreed non required defaults have been set to -1